### PR TITLE
Add CI tests for subpackages

### LINKED ISSUE #961: Add the CI tests for subpackages
Currently Travis and Appveyor only run tests for the top-level gfx_app package. Subpackages, such as gfx_core and gfx_window_* are not tested - only built. Travis and Appveyor scripts need to be changed to test all the packages in gfx repository. Cargo has the ability to run tests for a package given its spec: something like `cargo test -p gfx_core` will test the `gfx_core` package. Adding additional test commands to the scripts should be pretty simple.

### PR DESCRIPTION
**Pull request by eugene2k**
Changes to appveyor and travis scripts to include testing of submodules.
Changes to appveyor script to only download and install mingw when targetting mingw toolchain.
Code example moved from factory documentation into gfx_window_glutin.
Removed an outdated test, added no_run to some tests to prevent them from failing on CI builds.

fixes #961

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
-sudo: false
+sudo: required
```
Comment (kvark) [inline .travis.yml]: why is this required now?

**Comment by kvark** [comment]
What's up with all the whitespace?

File: .travis.yml (hunk 1 above)
Comment (eugene2k) [inline .travis.yml]: Can't add the sdl repository otherwise. It's not whitelisted.

**Comment by eugene2k** [comment]
Formatted the files while I was working on them and forgot it would show up in diffs. Is that an issue?

**Comment by kvark** [comment]
@homu r+
